File: sdk/runanywhere-swift/Sources/RunAnywhere/Core/Models/ModelFormat.swift
import Foundation

/// Model formats supported
public enum ModelFormat: String, CaseIterable {
    case mlmodel = "mlmodel"
    case mlpackage = "mlpackage"
    case tflite = "tflite"
    case onnx = "onnx"
    case ort = "ort"
    case safetensors = "safetensors"
    case gguf = "gguf"
    case ggml = "ggml"
    case mlx = "mlx"
    case pte = "pte"
    case bin = "bin"
    case weights = "weights"
    case checkpoint = "checkpoint"
    case unknown = "unknown"
}

File: sdk/runanywhere-android/src/main/java/com/runanywhere/sdk/models/ModelFormat.kt
package com.runanywhere.sdk.models

/**
 * Model formats supported
 */
enum class ModelFormat {
    TFLITE,
    ONNX,
    ORT,
    SAFETENSORS,
    GGUF,
    GGML,
    PTE,
    BIN,
    WEIGHTS,
    CHECKPOINT,
    UNKNOWN;
    
    companion object {
        fun fromString(value: String): ModelFormat {
            return values().find { it.name.equals(value, ignoreCase = true) } ?: UNKNOWN
        }
    }
} 